pinecone/data/index.py

```diff
@@ -197,81 +183,11 @@ def _upsert_batch(
-
-                indices = item["sparse_values"].get("indices", None)
-                values = item["sparse_values"].get("values", None)
-
-                if isinstance(values, np.ndarray):
-                    upsert_numpy_deprecation_notice("Deprecated type passed in sparse_values['values'].")
-                    values = values.tolist()
-                if isinstance(indices, np.ndarray):
-                    upsert_numpy_deprecation_notice("Deprecated type passed in sparse_values['indices'].")
-                    indices = indices.tolist()
-                try:
-                    item["sparse_values"] = SparseValues(indices=indices, values=values)
-                except TypeError as e:
-                    raise ValueError(
-                        "Found unexpected data in column `sparse_values`. "
-                        "Expected format is `'sparse_values': {'indices': List[int], 'values': List[float]}`."
-                    ) from e
-
-            if "metadata" in item:
-                metadata = item.get("metadata")
-                if not isinstance(metadata, Mapping):
-                    raise TypeError(f"Column `metadata` is expected to be a dictionary, found {type(metadata)}")
-
-            if isinstance(item["values"], np.ndarray):
-                upsert_numpy_deprecation_notice("Deprecated type passed in 'values'.")
-                item["values"] = item["values"].tolist()
-
-            try:
-                return Vector(**item)
-            except TypeError as e:
-                # if not isinstance(item['values'], Iterable) or not isinstance(item['values'][0], numbers.Real):
-                #     raise TypeError(f"Column `values` is expected to be a list of floats")
-                if not isinstance(item["values"], Iterable) or not isinstance(item["values"][0], numbers.Real):
-                    raise TypeError(f"Column `values` is expected to be a list of floats")
-                raise
-
-        def _vector_transform(item: Union[Vector, Tuple]):
-            if isinstance(item, Vector):
-                return item
-            elif isinstance(item, tuple):
-                if len(item) > 3:
-                    raise ValueError(
-                        f"Found a tuple of length {len(item)} which is not supported. "
-                        f"Vectors can be represented as tuples either the form (id, values, metadata) or (id, values). "
-                        f"To pass sparse values please use either dicts or a Vector objects as inputs."
-                    )
-                id, values, metadata = fix_tuple_length(item, 3)
-                return Vector(id=id, values=values, metadata=metadata or {}, _check_type=_check_type)
-            elif isinstance(item, Mapping):
-                return _dict_to_vector(item)
-            raise ValueError(f"Invalid vector value passed: cannot interpret type {type(item)}")
+        vec_builder = lambda v: VectorFactory.build(v, check_type=_check_type)
```

Comment:
`VectorFactory.build` is static, but we need to wrap it in this lambda to thread through the `check_type` param when mapping over all the vectors passed to upsert.